login/config.h

```diff
@@ -51,6 +51,9 @@ typedef struct glome_login_config {
 
   // Explicitly set host-id to use in the login request.
   const char* host_id;
+
+  // Private data for use by login operations (e.g. storing pam handle).
+  void* priv;
```

Comment:
I'm not a big fan of this, it looks like an inversion of responsibilities. At the very least, I'd want to make this dedicated to a specific login method (i.e. `pam_t* pam_handle;  // context info iff using PAM`).

Reply:
Yes, this is not great, but using `pam_t` in `config.h` would require PAM library for building `glome-login` binary too?

Reply:
Removed `priv` pointer and pass `pamh` directly now.